### PR TITLE
[Google Blockly] No modal input on mobile for FieldTextInput

### PR DESCRIPTION
**Pull request by bencodeorg**
We observed while bug bashing Google Blockly Dance Party on iPad that a new modal appeared when editing text fields (eg, the number in "after 4 measures" here:

https://studio.code.org/s/dance-2019/lessons/1/levels/4?blocklyVersion=Google

We also noticed that, after opening this modal, audio would no longer play after you hit "Run" until you refreshed the page.

This PR removes overrides the mobile-specific modal for all FieldTextInput fields (and applies this override to subclasses that are listed in the [Blockly docs](https://developers.google.com/blockly/reference/js/blockly)).

Also does a little refactor to make sure that every time we unregister/register a field for core Blockly, we also make it accessible to our code on the wrapper.

Strings representing each class in the field registry weren't listed in Google Blockly's docs, but Mike looked them up by searching for `fieldRegistry.register('` in their repo (thanks Mike!)

## Links

- jira ticket: [SL-364](https://codedotorg.atlassian.net/browse/SL-364)
- jira ticket: [SL-363](https://codedotorg.atlassian.net/browse/SL-363)

## Testing story

I tested via ngrok that the issue we observed on iPad no longer repros after this change on the level 4 link above. The field in "after 4 measures" is actually a `FieldNumber`, which subclasses `FieldTextInput`. I found an actual `FieldTextInput` on the project level for Dance (/projects/dance/new, with the google blockly URL param) in the name of a function block (by default, "do something") -- I was able to edit this without issue on my laptop/iPad.

### CODE REVIEW

File: apps/src/blockly/addons/cdoFieldVariable.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import msg from '@cdo/locale';
 const RENAME_THIS_ID = 'RENAME_THIS_ID';
 const RENAME_ALL_ID = 'RENAME_ALL_ID';
 
-export default class FieldVariable extends GoogleBlockly.FieldVariable {
+export default class CdoFieldVariable extends GoogleBlockly.FieldVariable {
```
Comment (bencodeorg) [inline apps/src/blockly/addons/cdoFieldVariable.js:7]: These class renames are just for standardization/convenience when debugging -- we were renaming these on import to googleBlocklyWrapper to CdoFieldVariable, but the Chrome debugger shows the actual class name, which makes distinguishing them from google blockly difficult.

Comment (mikeharv) [reply apps/src/blockly/addons/cdoFieldVariable.js:7]: I like this change/convention!

File: apps/src/blockly/addons/cdoFieldAngle.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import GoogleBlockly from 'blockly/core';
+import {noMobileShowEditorOverride} from './cdoFieldTextInput';
+
+export default class CdoFieldAngle extends GoogleBlockly.FieldAngle {
+  showEditor_ = noMobileShowEditorOverride;
+}
```
Comment (mikeharv) [inline apps/src/blockly/addons/cdoFieldAngle.js:6]: Thanks for adding these other classes!

File: apps/src/blockly/googleBlocklyWrapper.js [hunk 3]
```diff
@@ -62,6 +67,26 @@ const BlocklyWrapper = function(blocklyInstance) {
       }
     });
   };
+
+  /**
+   * Override core Blockly fields with Code.org customized versions,
+   * and sets the field on our wrapper for use by our code.
+   * @param {array} overrides (elements are arrays of shape [fieldRegistryName, fieldClassName, fieldClass])
+   */
+  this.overrideFields = function(overrides) {
```
Comment (mikeharv) [inline apps/src/blockly/googleBlocklyWrapper.js:76]: Neat! I don't have a strong opinion here, but what would you think about moving this to cdoUtils?

**Review by mikeharv — APPROVED** [review]
This looks great, Ben! Thanks for taking care of all the other classes that extend FieldTextInput.

File: apps/src/blockly/googleBlocklyWrapper.js (hunk 3 above)
Comment (bencodeorg) [reply apps/src/blockly/googleBlocklyWrapper.js:76]: mmm I'd vote to keep it here for a couple reasons:
- it's quite similar functionally to the `wrapReadOnlyProperty` and `wrapSettableProperty` functions defined here, so I think keeping them together makes sense.
- it seems like cdoUtils is useful for situations where we want to use the same function name whether we're using cdo or google blockly (eg, when your recommended it [here](https://github.com/code-dot-org/code-dot-org/pull/48744/files)), and/or if we want to reuse functions across multiple files. I don't think either case applies here? Not sure I have a perfect understanding of what cdoUtils is explicitly for though :)

Comment (mikeharv) [reply apps/src/blockly/googleBlocklyWrapper.js:76]: Both of those points make sense to me. Let's keep it here. :)

**Pull request merged**